presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -368,8 +370,18 @@ private boolean isCandidate(Map<ColumnHandle, NullableValue> bindings)
             }
             LookupSymbolResolver inputs = new LookupSymbolResolver(assignments, bindings);
 
+            Object optimized;
+            try {
+                optimized = evaluator.optimize(inputs);
+            }
+            catch (PrestoException e) {
+                // Skip pruning if evaluation fails in a recoverable way.
+                // Failing here can cause spurious query failures for empty partitions.
```

Comment:
Would it be possible to have a test?

Reply:
Do you have thoughts on how to do this? Let's me see if I can create a Hive integration test.